### ISSUE TITLE
java.lang.NullPointerException

### ISSUE THREAD

**Issue by boris-noyvert** [TeWu/GmailAttachmentsExtractor#5]
GmailAttachmentsExtractor crashes on some emails. It does download attachments, but the copy email without attachments is not created, and the whole run is stopped. If I run the same command again it will crash on exactly the same email. I don't see any pattern in the emails that cause the crash.

```
72/74 (97%) | Processing email 'Re: 10 March ?'
    Extracting 2 attachment(s) to directory '2008.02.06 18_51_17 Re_ 10 March _'
    Attachment saved: KTE_FINAL.pdf
    Attachment saved: KText.pdf
    Inserting copy of email without extracted attachments to Gmail
java.lang.NullPointerException
        at pl.geek.tewu.gmail_attachments_extractor.GmailAttachmentsExtractor.extractAttachments(GmailAttachmentsExtractor.java:196)
        at pl.geek.tewu.gmail_attachments_extractor.Main.call(Main.java:45)
        at pl.geek.tewu.gmail_attachments_extractor.Main.call(Main.java:14)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1783)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2150)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2144)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1975)
        at picocli.CommandLine.execute(CommandLine.java:1904)
        at pl.geek.tewu.gmail_attachments_extractor.Main.main(Main.java:29)
```

Also as a general idea, it would be good to introduce some error handling, so that the run doesn't stop on an error, but continues processing the remaining emails.

Thank you!

**Comment by TeWu** [comment]
Thanks for the info.
Does #4 solve your problem?

Error handling is that way for a reason. Since GmailAttachmentsExtractor can be used to archive some important data, it would be really bad if it would fail to do that without giving the user impossible to miss information that it has failed. It is far more important for me to not expose someone to the risk of data loss, than making the program usable for everyone in every situation. Anyway - thanks for the feedback.

**Comment by boris-noyvert** [comment]
Yes, this is the same issue. Applying a label to the email before run fixes it.
(Sorry, I should have checked other reported issues first.)
I will wait for the next release.
Many thanks!

**Comment by boris-noyvert** [comment]
Yes, this is the same issue. Applying a label to the email before run fixes it.
(Sorry, I should have checked other reported issues first.)
I will wait for the next release.
Many thanks!

**Comment by TeWu** [comment]
@boris-noyvert FYI, I've just released [v1.0.2](https://github.com/TeWu/GmailAttachmentsExtractor/releases/tag/1.0.2).
